go/vt/vtgate/planbuilder/select2.go

```diff
@@ -0,0 +1,200 @@
+// Copyright 2016, Google Inc. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package planbuilder
+
+import (
+	"encoding/json"
+	"errors"
+
+	"github.com/youtube/vitess/go/vt/sqlparser"
+)
+
+// PlanBuilder represents any object that's used to
+// build a plan.
+type PlanBuilder interface{}
+
+// JoinBuilder is used to build a Join primitive.
+type JoinBuilder struct {
+	IsLeft      bool
+	Left, Right PlanBuilder
+}
+
+// RouteBuilder is used to build a Route primitive.
+type RouteBuilder struct {
+	From     sqlparser.TableExpr
+	ChunkNum int
```

Comment:
You're currently not setting this anywhere. I'm assuming you'll use this in the future. Just thought I'd mention in case it's a mistake.